### ISSUE TITLE
[GSoC] Skeleton Tool 

### ISSUE THREAD

**Issue by 9inpachi** [synfig/synfig#759]
Hi. My name is Fawad Ali and I would like to code in this year's GSoC for this Open Source software.
I have a personal interest in animation, especially Anime. I have worked in Animate CC and made some simple animations for a couple of games (frame-by-frame mostly) and I have prior experience with C++ and Visual C++.
I would like to work on the Skeleton Tool project for the summer if I am selected.
It would be helpful if you could tell me what I can possibly do before the proposal submission to have a higher probability of getting selected.
I have fixed one issue so far and would like to get more done.
Is there any issue that would be related to this project (skeleton tool) through which I can have a better understanding?
Anything that helps me to better understand the implementation of this will be very helpful.
Thanks. ^_^

**Comment by Jose-Moreno** [comment]
@9inpachi Hi. Although I'm not part of Synfig's team, I can tell you in advance that there are some important issues related to the Skeleton that range from critical bugs to low priority enhancement requests, which i'll outline as follows:

**Bugs**

- [Unexpected bone duplication when opening files from previous versions](https://github.com/synfig/synfig/issues/714)
- [Artwork origin point changing with bone angle](https://github.com/synfig/synfig/issues/632)
- [Skeleton copy references original bones between files](https://github.com/synfig/synfig/issues/455)
- [Crash on removing parent bone](https://github.com/synfig/synfig/issues/375)
- [Unexpected deformation quality with skeleton on vector artwork](https://github.com/synfig/synfig/issues/345)

**Requests**

- [Implement "Draw Skeleton" Tool](https://github.com/synfig/synfig/issues/602)
- [Implement Mesh-based Skeletal Deformation](https://github.com/synfig/synfig/issues/558)
- [Skeleton Independent Copy (non-referenced)](https://github.com/synfig/synfig/issues/453) **Note**: This is related to Issue 455
- [Implement Bone Dynamics](https://github.com/synfig/synfig/issues/312)

**Related Plugins**

- [Incorrect limb bone label on skeleton plugin](https://github.com/synfig/synfig/issues/685)

There is also a ["Bones" label](https://github.com/synfig/synfig/issues?q=is%3Aissue+is%3Aopen+skeleton+label%3ABones), for issues related to the skeleton tool which contains some of the points on this list, but requires updating.

I hope this helps you get started while you obtain official respond from @morevnaproject

**Comment by 9inpachi** [comment]
Thanks @Jose-Moreno.
I will try looking into one of these issues. Hope I can get a good grasp.

**Comment by morevnaproject** [comment]
@Jose-Moreno Thank you for giving the summary of current issues!

@9inpachi Hi Fawad! I am happy you got interested in this project! ^__^
Before we start discussing, it is important to make sure you understand the current functionality of bones in Synfig. The best way to achieve this is to watch our video course. Please mail me via https://www.synfig.org/contact/ and I will provide you a personal link for free access to the course.

**Comment by Gilraiser** [comment]
Hello, I hope you are very well. I want to send you a better example of the error that causes some bones to double,  I hope this information helps the project. Thank you.

This is the initial file. 
[example-01.sif.zip](https://github.com/synfig/synfig/files/3537588/example-01.sif.zip)

This is after copying it to another file.
[example-02.sifz.zip](https://github.com/synfig/synfig/files/3537589/example-02.sifz.zip)

Here I attach a video demonstration.
https://www.youtube.com/watch?v=aTBuhxQpS1c&feature=youtu.be

**Comment by 9inpachi** [comment]
Hi Gillraiser,

Thanks for the suggestions. This project was not initiated in GSoC.
I think the errors you mention relate to the bones already implemented.
@morevnaproject will have a better understanding of this.

Cheers. :)

**Comment by Gilraiser** [comment]
Ok, thank you very much.
